Source/LightsOutSln/LightsOut/Patches/ModCompatibility/Biotech/PatchMechCharger.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using HarmonyLib;
+using LightsOut.Common;
+using LightsOut.Patches.Power;
+using RimWorld;
+using Verse;
+
+namespace LightsOut.Patches.ModCompatibility.Biotech
+{
+    /// <summary>
+    /// Disable power draw on mech recharger if not in use by mech
+    /// </summary>
+    public class PatchMechCharger : ICompatibilityPatchComponent<PatchMechCharger>
+    {
+        public override string ComponentName => "Patch for mech recharger power draw";
+
+        public override IEnumerable<PatchInfo> GetPatches(Type type)
+        {
+            var patches = BiotechCompatibilityPatch.CustomOnOffPatches(GetMethod<Building_MechCharger>("StartCharging"),
+                GetMethod<Building_MechCharger>("StopCharging"));
+            
+            patches.Add(
+                new PatchInfo
+                {
+                    method = GetMethod(typeof(Tables), nameof(Tables.IsTable)),
+                    patch = GetMethod<PatchMechCharger>(nameof(Post_IsTable)),
+                    patchType = PatchType.Postfix
+                }
+            );
+            patches.Add(
+                new PatchInfo
+                {
+                    method = GetMethod(typeof(Building), nameof(Building.SpawnSetup)),
+                    patch = GetMethod<PatchMechCharger>(nameof(AfterSpawn)),
+                    patchType = PatchType.Postfix
+                }
+            );
+            return patches;
+        }
+
+        private static PropertyInfo IsAttachedToMech = null;
+
+        private static void AfterSpawn(Building __instance)
+        {
+            if (IsAttachedToMech == null)
```

Comment:
Realistically I could go either way on this since it's probably a micro-optimization, but having this check here guarantees that we do the ref equals check on this property for every building spawn in the game. If you move this if statement inside of the __instance is Building_MechCharger condition then we will only pay for it for the mech chargers.

Better yet, you could start this method out with `if (!(__instance is Building_MechCharger ch)) { return; }` so that everything else exits as early as possible.